src/Engine/ProtoCore/SyntaxAnalysis/ImperativeAstVisitor.cs

```diff
@@ -345,6 +346,82 @@ public virtual TResult VisitIfStmtPositionNode(IfStmtPositionNode node)
         }
     }
 
+    internal class ImperativeIdentifierInPlaceMapper : ImperativeAstReplacer
+    {
+        private Core core;
+        private Func<string, bool> cond;
+        private Func<string, string> mapper;
+
+        public ImperativeIdentifierInPlaceMapper(Core core, Func<string, bool> cond, Func<string, string> mapper)
+        {
+            this.core = core;
+            this.cond = cond;
+            this.mapper = mapper;
+        }
+
+        public override ImperativeNode VisitLanguageBlockNode(LanguageBlockNode node)
+        {
+            var cbn = node.CodeBlockNode as CodeBlockNode;
+            if (cbn == null)
+            {
+                var assoc_cbn = node.CodeBlockNode as AST.AssociativeAST.CodeBlockNode;
```

Comment:
so this handles associative blocks inside of imperative blocks, what about the other way around?

Reply:
Ah, good catch! Will make a similar change for the associative class.

Reply:
I had made the change but had commented it out for some reason and then cleaned up the commented code 🤷‍♂️